game/tower/army-tower/army-tower.js

```diff
@@ -0,0 +1,153 @@
+const { getNewTowerItem, getTowerItem } = require("../../../game/items/tower-items/tower-item-functions");
+// Takes an array of users and makes them fight together in the Tower
+// Category is "solo" or "trio" etc
+const armyTowerFight = async (users, category) => {
+	// The results sent to the results embed after the combat
+	const allResults = {
+		users,
+		combatResults: [],
+		drops: [],
+		enemy: {},
+		highestLevel: 0,
+		newLevel: 0,
+		won: false,
+	};
+
+	const highestLevel = users.reduce((acc, cur) => {
+		return cur.tower[`${category} full-army`].level > acc ? cur.tower[`${category} full-army`].level : acc;
+	}, 0);
+
+	const originalEnemy = getArmyTowerEnemies(highestLevel);
+	const enemy = { ...originalEnemy, stats: { ...originalEnemy.stats } };
+
+	allResults.enemy = enemy;
+
+	const enemyCombatModifier = highestLevel * 0.8 * (1 + Math.random() * 0.5);
+
+	for(const stat in enemy.stats) {
+		enemy.stats[stat] = Math.floor(enemy.stats[stat] * enemyCombatModifier);
+	}
+
+	const combatResults = users.map(user => calculatePveFullArmyResult(user, enemy));
+
+	// Make the users lose hero hp and units depending on lossPercentage and give exp
+	users.forEach(user => {
+		const userCombatResults = combatResults.find(cr => cr.userId === user.account.userId);
+		// user.setNewCooldown("tower", new Date());
+		user.unitLoss(userCombatResults.lossPercentage, true);
+		user.alternativeGainXp(userCombatResults.expReward);
+	});
+
+	// While loop to iterate over combat results and let the other users fight the remaining forces
+	let losingCombatResults = combatResults.filter(cr => !cr.win);
+	const winningCombatResults = combatResults.filter(cr => cr.win);
+
+
+	let healthLeft = users.filter(user => healthLeftOnArmy(user)).length > 0;
+
+	// This has not been tested yet. It is used when there are several players attending the fight. Lets the remaining players fight the remaining enemies.
+	while(!(losingCombatResults.length === 0 || !healthLeft)) {
+		losingCombatResults = losingCombatResults.map(cr => {
+			const remainingUsers = users.filter(user => healthLeftOnArmy(user));
+			const randomUser = Math.floor(Math.random() * remainingUsers.length);
+
+			const newEnemy = { ...enemy };
+			const statsLength = Object.keys(newEnemy.stats).length;
+			for(const stat in newEnemy.stats) {
+				newEnemy[stat] = newEnemy[stat] - Math.floor(cr.remainingForces / statsLength);
+			}
+
+			console.log("NEW ENEMIES", newEnemy, enemy);
+
+			const remainingFightResult = calculatePveFullArmyResult(remainingUsers[randomUser]);
+			combatResults.push(remainingFightResult);
+
+			if(remainingFightResult.win) {
+				winningCombatResults.push(remainingFightResult);
+				return false;
+			}
+			else {
+				return remainingFightResult;
+			}
+		}).filter(el => el);
+
+		const remainingUsers = users.filter(user => healthLeftOnArmy(user));
+
+		healthLeft = remainingUsers.filter(user => healthLeftOnArmy(user)).length > 0;
+		console.log("HEALTH LEFT", healthLeft);
```

Comment:
console.log

Reply:
Tenkte å la de ligge, jeg kan sikkert kommentere dem ut, men trenger dem til trio/duo tower, og er redd for å glemme det.

Reply:
ok